GVFS/GVFS.Common/FileSystem/PhysicalFileSystem.cs

```diff
@@ -36,7 +36,17 @@ public virtual void DeleteDirectory(string path, bool recursive = true)
                 }
             }
 
-            directory.Delete();
+            try
+            {
+                directory.Delete();
+            }
+            catch (Exception)

```

Comment:
Can we catch something more specific, like `IOException`?

Reply:
I thought about that but in the [git code](https://github.com/microsoft/git/blob/a925f8273bfa007717dbfe392a9b006fe89e51fd/symlinks.c#L281), they aren't checking for any specific type of error.  They simply leave the directory there.

The [other possible exceptions](https://docs.microsoft.com/en-us/dotnet/api/system.io.directoryinfo.delete?view=netframework-4.8) `UnauthorizedAccessException` and `SecurityException` seem like we would want to keep going on these as well since this should be an empty folder and it's not going to hurt anything if it is left out there, but if we do fail dehydrate then there could be files that will show as deleted the next `git status` since we don't remove the directory from the modified paths and this is the issue we are trying to avoid.

Reply:
Those extra reception types make sense. Thanks!